src/vaults/locked/StrategyVaultV2.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.16;
+
+import {MathUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/math/MathUpgradeable.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+
+import {StrategyVault} from "src/vaults/locked/StrategyVault.sol";
+import {NftGate} from "src/vaults/NftGate.sol";
+import {HarvestStorage} from "src/vaults/HarvestStorage.sol";
+import {VaultErrors} from "src/libs/VaultErrors.sol";
+
+contract StrategyVaultV2 is StrategyVault, NftGate, HarvestStorage {
+    using SafeTransferLib for ERC20;
+    using MathUpgradeable for uint256;
+
+    function _deposit(address caller, address receiver, uint256 assets, uint256 shares) internal virtual override {
+        _checkNft(receiver);
+        if (shares == 0) revert VaultErrors.ZeroShares();
+        _mint(receiver, shares);
+        _asset.safeTransferFrom(caller, address(this), assets);
+        emit Deposit(caller, receiver, assets, shares);
+    }
+
+    function _getWithdrawalFee(uint256 assets, address owner) internal view virtual override returns (uint256) {
```

Comment:
Every V2 has this similar implementation. Can we move this to a common contract.

Reply:
Different storage layouts prevent this (if we inherit from the contract). We could make a `WithdrawalFeeModule` though. I think it's only worth it to do that to get under the code size limit.